test_pyproject_support and setuptools 61.0.0

With `setuptools` 61.0.0 `test_pyproject_support` fails:
```
====================================== FAILURES ======================================
_______________________________ test_pyproject_support _______________________________

tmpdir = local('/usr/src/tmp/pytest-of-builder/pytest-9/test_pyproject_support0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fd87262efb0>

    def test_pyproject_support(tmpdir, monkeypatch):
        pytest.importorskip("toml")
        monkeypatch.delenv("SETUPTOOLS_SCM_DEBUG")
        pkg = tmpdir.ensure("package", dir=42)
        pkg.join("pyproject.toml").write_text(
            textwrap.dedent(
                """
                [tool.setuptools_scm]
                fallback_version = "12.34"
                [project]
                description = "Factory ⸻ A code generator 🏭"
                authors = [{name = "Łukasz Langa"}]
                """
            ),
            encoding="utf-8",
        )
        pkg.join("setup.py").write("__import__('setuptools').setup()")
        res = do((sys.executable, "setup.py", "--version"), pkg)
>       assert res == "12.34"
E       AssertionError: assert 'DESCRIPTION:...     }\n    }' == '12.34'
E         - 12.34
E         + DESCRIPTION:
E         +     Data structure for the **project** table inside ``pyproject.toml`` (as
E         +     initially defined in :pep:`621`)
E         + 
E         + GIVEN VALUE:
E         +     {...
E         
E         ...Full output truncated (489 lines hidden), use '-vv' to show

testing/test_integration.py:41: AssertionError
-------------------------------- Captured stdout call --------------------------------
/usr/src/RPM/BUILD/setuptools_scm/.tox/py310-test/lib/python3/site-packages/setuptools/config/pyprojecttoml.py:102: _ExperimentalProjectMetadata: Support for project metadata in `pyproject.toml` is still experimental and may be removed (or change) in future releases.
  warnings.warn(msg, _ExperimentalProjectMetadata)
configuration error: `project` must contain ['name'] properties
Traceback (most recent call last):
  File "/usr/src/tmp/pytest-of-builder/pytest-9/test_pyproject_support0/package/setup.py", line 1, in <module>
    __import__('setuptools').setup()
  File "/usr/src/RPM/BUILD/setuptools_scm/.tox/py310-test/lib/python3/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/usr/src/RPM/BUILD/setuptools_scm/.tox/py310-test/lib/python3/site-packages/setuptools/_distutils/core.py", line 122, in setup
    dist.parse_config_files()
  File "/usr/src/RPM/BUILD/setuptools_scm/.tox/py310-test/lib/python3/site-packages/_virtualenv.py", line 21, in parse_config_files
    result = old_parse_config_files(self, *args, **kwargs)
  File "/usr/src/RPM/BUILD/setuptools_scm/.tox/py310-test/lib/python3/site-packages/setuptools/dist.py", line 855, in parse_config_files
    pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
  File "/usr/src/RPM/BUILD/setuptools_scm/.tox/py310-test/lib/python3/site-packages/setuptools/config/pyprojecttoml.py", line 54, in apply_configuration
    config = read_configuration(filepath, True, ignore_option_errors, dist)
  File "/usr/src/RPM/BUILD/setuptools_scm/.tox/py310-test/lib/python3/site-packages/setuptools/config/pyprojecttoml.py", line 120, in read_configuration
    validate(subset, filepath)
  File "/usr/src/RPM/BUILD/setuptools_scm/.tox/py310-test/lib/python3/site-packages/setuptools/config/pyprojecttoml.py", line 43, in validate
    raise error from None
ValueError: invalid pyproject.toml config: `project`
```

https://setuptools.pypa.io/en/latest/history.html#v61-0-0:
> pypa/setuptools#3067: If the file pyproject.toml exists and it includes project metadata/config (via [project] table
   or [tool.setuptools]), a series of new behaviors that are not backward compatible may take place:
> - The default value of include_package_data will be considered to be True.
> - Setuptools will attempt to validate the pyproject.toml file according to PEP 621 specification.
> - The values specified in pyproject.toml will take precedence over those specified in setup.cfg or setup.py.

According to https://packaging.python.org/en/latest/specifications/declaring-project-metadata/:
> The only fields required to be statically defined are:
> - name
>
> The fields which are required but may be specified either statically or listed as dynamic are:
> - version

Upstrem didn't catch this because of the wrong skip condition `pytest.importorskip("toml")`, while `toml` is no longer supported.
Note: `toml` may be pulled indirectly to test environment.

--- a/testing/test_integration.py
+++ b/testing/test_integration.py
@@ -21,7 +21,7 @@ def wd(wd):
 
 
 def test_pyproject_support(tmpdir, monkeypatch):
-    pytest.importorskip("toml")
+    pytest.importorskip("tomli")
     monkeypatch.delenv("SETUPTOOLS_SCM_DEBUG")
     pkg = tmpdir.ensure("package", dir=42)
     pkg.join("pyproject.toml").write_text(
@@ -30,8 +30,10 @@ def test_pyproject_support(tmpdir, monkeypatch):
             [tool.setuptools_scm]
             fallback_version = "12.34"
             [project]
+            name = "foo"
             description = "Factory ⸻ A code generator 🏭"
             authors = [{name = "Łukasz Langa"}]
+            dynamic = ["version"]
             """
         ),
         encoding="utf-8",

